### PR TITLE
build(fluid-static): use fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
Blocked on https://github.com/microsoft/FluidFramework/pull/19832

Semi-automatic PR to switch to fluid-tsc

### CODE REVIEW

File: packages/framework/fluid-static/package.json [hunk 1]
```diff
@@ -89,9 +71,11 @@
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../.prettierignore",
 		"test": "npm run test:mocha",
 		"test:coverage": "c8 npm test",
-		"test:mocha": "mocha --recursive \"dist/test/**/*.spec.*js\" -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha": "npm run test:mocha:cjs && npm run test:mocha:esm",
+		"test:mocha:cjs": "mocha  --recursive \"dist/test/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha:esm": "mocha  --recursive \"lib/test/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
```
Comment (jason-ha) [inline packages/framework/fluid-static/package.json:76]: nit: there are two spaces in between mocha and --recursive.

File: packages/framework/fluid-static/tsconfig.json [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
-		"../../../common/build/build-common/tsconfig.cjs.json",
-	],
+	"extends": "../../../common/build/build-common/tsconfig.node16.json",
 	"include": ["src/**/*"],
 	"exclude": ["src/test/**/*"],
 	"compilerOptions": {
 		"rootDir": "./src",
-		"outDir": "./dist",
+		"outDir": "./lib",
 		"types": ["node"],
```
Comment (jason-ha) [inline packages/framework/fluid-static/tsconfig.json:8]: pre-existing but this does not look right. I don't think fluid-static should import node types.

**Review by jason-ha — APPROVED** [review]
We'll need to watch node dependencies. But perhaps that should be a follow up.

**Comment by jason-ha** [comment]
api-extractor-esm.json is now unused and should be removed. also can be a post fix.

File: packages/framework/fluid-static/package.json (hunk 1 above)
Comment (DLehenbauer) [reply packages/framework/fluid-static/package.json:76]: Thx.  I fixed this in my template as well.

File: packages/framework/fluid-static/tsconfig.json (hunk 2 above)
Comment (DLehenbauer) [reply packages/framework/fluid-static/tsconfig.json]: Removed, thx.

**Comment by DLehenbauer** [comment]
The conversion script deletes 'api-extractor-esm.json', but the new '-cjs' file is similar enough that GitHub is displaying it as a rename rather than a removal.

**Review by jason-ha — APPROVED** [review]

**Pull request merged**